prow/sidecar/censor.go

```diff
@@ -260,7 +260,9 @@ func validRelPath(p string) bool {
 
 // archive re-packs the dir into the destination
 func archive(srcDir, destArchive string) error {
-	output, err := ioutil.TempFile("", "tmp-archive")
+	// we want the temporary file we use for output to be in the same directory as the real destination, so
+	// we can be certain that our final os.Rename() call will not have to operate across a device boundary
+	output, err := os.Create(fmt.Sprintf("%s.tmp-archive", destArchive))
```

Comment:
why not:
```suggestion
	output, err := ioutil.TempFile(filepath.Dir(destArchive), "tmp-archive"))
```
`TempFile` can use the dest dir but still retain the non-conflicting random filename suffix behavior

Reply:
Ah, gotcha - misread the `ioutil.Tempdir` doc, thought it put stuff under `$TMP` no matter what